executor/builder.go

```diff
@@ -975,16 +974,16 @@ func (b *executorBuilder) buildIndexLookUpJoin(v *plan.PhysicalIndexJoin) Execut
 	}
 }
 
-func (b *executorBuilder) buildTableReader(v *plan.PhysicalTableReader) *TableReaderExecutor {
-	dagReq := b.constructDAGReq(v.TablePlans)
-	if b.err != nil {
-		return nil
+func buildTableReader(b *executorBuilder, v *plan.PhysicalTableReader) (*TableReaderExecutor, error) {
+	dagReq, err := constructDAGReq(v.TablePlans, b.getStartTS(), b.ctx)
+	if err != nil {
+		return nil, errors.Trace(err)
 	}
 	ts := v.TablePlans[0].(*plan.PhysicalTableScan)
 	table, _ := b.is.TableByID(ts.Table.ID)
-	newRanges := b.constructTableRanges(ts)
-	if b.err != nil {
-		return nil
+	newRanges, err := constructTableRanges(ts, b.ctx)
```

Comment:
We can move the range construction out to `executorBuilder.buildTableReader`, we don't need range from the plan in `dataReaderBuilder`.